changelogs/unreleased/stdout_log_referenced_before_assignment.yml

```
@@ -0,0 +1,8 @@
+---
+description: "Fix bug where starting the scheduler can result in an error saying that stdout_log is referenced before assignment. This can cause the compiler service to get stuck or trigger the agent deploy timers at incorrect moments."
```

The user doesn't know about the scheduler.
```suggestion
description: "Fix bug where starting the agent can result in an error saying that stdout_log is referenced before assignment. This can cause the compiler service to get stuck or trigger the agent deploy timers at incorrect moments."
```